xapian-core/backends/glass/glass_changes.cc

```diff
@@ -63,8 +64,10 @@ GlassChanges::start(glass_revision_number_t old_rev,
 
     // Always check max_changesets for modification since last revision.
     const char *p = getenv("XAPIAN_MAX_CHANGESETS");
-    if (p) {
-	max_changesets = atoi(p);
+    if (p && *p) {
+	if (!parse_unsigned(p, max_changesets)) {
+	    throw "XAPIAN_MAX_CHANGESETS must be a non-negative integer";
```

Comment:
I did some final testing, and this is wrong:

```
$ XAPIAN_MAX_CHANGESETS=-1 examples/simpleindex tmp.db
terminate called after throwing an instance of 'char const*'
Aborted
```

We shouldn't be throwing `const char*` exceptions from the library - probably `Xapian::InvalidArgumentError` is the most appropriate option here.

Reply:
fixed in a fixup commit.